pkg/dns/server_test.go

```diff
@@ -121,5 +124,30 @@ var _ = Describe("DNS server", func() {
 			// and metrics are published
 			Expect(test_metrics.FindMetric(metrics, "dns_server_resolution", "result", "unresolved").Counter.GetValue()).To(Equal(1.0))
 		})
+
+		It("should resolve services with '.'", func() {
+			// given
+			dnsResolver.SetVIPs(vips.List{
+				"my.service": "240.0.0.1",
+			})
```

Comment:
If you do this in the `It` let's remove it from `BeforeEach` and put it in each `It`.